src/core/compiler/CircomCompilerFactory.ts

```diff
@@ -1,44 +1,129 @@
-import fs from "fs";
+import os from "os";
+import path from "path";
+import fs from "fs-extra";
+import { promisify } from "util";
+import { exec } from "child_process";
 
-import { WASMCircomCompiler, NativeCircomCompiler } from "./CircomCompiler";
 import { HardhatZKitError } from "../../errors";
-import { execCall } from "../../utils/utils";
 
-import { ICircomCompiler, IWASMCircomCompiler, CompilerVersion } from "../../types/core";
+import { BinaryCircomCompiler, WASMCircomCompiler } from "./CircomCompiler";
+import { CompilerDownloader } from "./CircomCompilerDownloader";
+
+import { ICircomCompiler, IWASMCircomCompiler, NativeCompiler } from "../../types/core";
+import { CompilerPlatformBinary } from "../../types/core/compiler/circom-compiler-downloader";
+
+import { isVersionHigherOrEqual } from "../utils/VersionManagement";
 
 // eslint-disable-next-line
 const { Context } = require("@distributedlab/circom2");
 
 export class CircomCompilerFactory {
-  public static createWASMCircomCompiler(version: CompilerVersion): IWASMCircomCompiler {
-    switch (version) {
-      case "0.2.18":
-        return new WASMCircomCompiler(this._getCircomCompiler("@distributedlab/circom2/circom.wasm"));
-      default:
-        throw new HardhatZKitError(`Unsupported Circom compiler version - ${version}. Please provide another version.`);
+  public static createWASMCircomCompiler(highestVersion: string): IWASMCircomCompiler {
+    if (!isVersionHigherOrEqual("2.1.9", highestVersion)) {
```

Comment:
Please move this `2.1.9` magic number to constants.

Is [this](https://github.com/dl-solarity/hardhat-zkit/blob/feature/compiler-management/src/types/core/compiler/circom-compiler-factory.ts) used?